### DISCUSSION TITLE
Paste on Right Click

### DISCUSSION THREAD

**Discussion by ket000** [kovidgoyal/kitty#3923] [Q&A]
I enabled copy_on_select yes
I want to use right click to paste it instead of middle click.

I tried this

#mouse_map middle        release ungrabbed paste_from_selection
mouse_map right        release ungrabbed,grabbed paste_from_selection

#mouse_map right press ungrabbed mouse_selection extend
mouse_map right press ungrabbed mouse_selection extend no_op

It sill extends the selection instead of pasting it.

Please let me know How can I use right click to paste instead of left click.

My version is
kitty 0.22.2 created by Kovid Goyal
ketan@ketanzorinos:~$ 

It will be nice if we can emulate default putty behavior for kitty.

**Comment by kovidgoyal** [comment]
Go to
https://sw.kovidgoyal.net/kitty/conf/#mouse-actions

find the one where extend selection is defined. copy/paste it and change
the action to what you want.

And note that this kitty is no relation to putty.

**Reply by tjatjagithub** [reply]
This is not because of PuTTY, but in general most systems I worked with allow to select a text with the mouse and paste it in another windows with the right mouse button!

The middle mouse button is often problematic, as it is a mouse wheel too.

I tried what you suggested, but am clearly too stupid:

mouse_map right press ungrabbed mouse_selection paste_from_clipboard
mouse_map right press ungrabbed mouse_selection paste_from_selection

Both did not work .... 

And I also tried just those:

mouse_map right paste_from_clipboard
mouse_map right paste_from_selection
mouse_map right press paste_from_clipboard
mouse_map right press paste_from_selection

Could you help, please?
I want to get away from iTerm2 which is the only other App that seems to support this.

Many thanks!

**Comment by tjatjagithub** [comment]
I found a decent solution, @ket000 

```
detect_urls no
copy_on_select clipboard
strip_trailing_spaces smart
select_by_word_characters ${}/&@_.
focus_follows_mouse yes
mouse_map right press grabbed,ungrabbed no-op
mouse_map right click grabbed,ungrabbed paste_from_clipboard
allow_hyperlinks no

```
This quite usable, only the newline character does not get copied.
So you cannot copy-paste whole lines.

Also, there is that delay because of some other detection ... this slows everything down.
Will try to get it away to be able to copy-paste faster.

The best I could do, is with "click_interval 0.2" ...

But I fear that this is not fully possible with Kitty.
This delay is required for left button double or tripple clicks, but cannot be disabled or set to 0 for the right button.

**Comment by tjatjagithub** [comment]
Sadly, this still does not work.

Hyperlinks continue to be interpreted and open a webbrowser, so you cannot even copy-paste them!

No idea, how to fix this :-(

**Comment by ket000** [comment]
I did the following on $HOME/.config/kitty/kitty.conf and it works as I wanted. 


copy_on_select yes

#mouse_map right press ungrabbed mouse_selection extend
mouse_map right press ungrabbed paste_from_selection

**Reply by tjatjagithub** [reply]
Interesting.

So you just replaced the "press" event.

I started searching for "click" events and only later noticed "press".
Still not sure what is the difference between both.

Meanwhile, I fixed the URL problem:

```
# Do not handle URLs
mouse_map left click ungrabbed no-op

```

So, the only thing missing is copying the "newline" character, when pasting a while line.

**Reply by tjatjagithub** [reply]
Ohh!

You were so right, @ket000 !

The "press" event removes the timing problem of "click".
Fantastic, many thanks!

**Reply by tjatjagithub** [reply]
I do am wondering why "copy_on_select yes" works.

The syntax should either be "copy_on_select clipboard" or "copy_on_select name_of_buffer", with "name_of_buffer" something like "a1".

At least the documentation reads like this!

```
#: Copy to clipboard or a private buffer on select. With this set to
#: clipboard, simply selecting text with the mouse will cause the text
#: to be copied to clipboard. Useful on platforms such as macOS that
#: do not have the concept of primary selections. You can instead
#: specify a name such as a1 to copy to a private kitty buffer
#: instead. Map a shortcut with the paste_from_buffer action to paste
#: from this private buffer. For example::

#:     map cmd+shift+v paste_from_buffer a1

```

**Comment by ket000** [comment]
MY solution works with multi line also and no need to change URL handling etc. So this makes it behave like putty on windows. I will stick with this solution as I do not use extend. Feel free to use whatever works for you.

**Reply by tjatjagithub** [reply]
Could you post your whole configuration for this PuTTY behavior?
I needed to use those 3 to not handle URLs (strange idea for a terminal emulator):

```
mouse_map left click ungrabbed no-op
detect_urls no
allow_hyperlinks no
```

If I do not use them, URLs get marked when hovering over them and a webbrowser will be opened on left click.

**Reply by Ryder17z** [reply]
> I do not use extend. Feel free to use whatever works for you.

How exactly is it supposed to work? - because your solution doesn't work. Removing the # does make it work though.

**Comment by tjatjagithub** [comment]
Also, I just now noticed this comment:

```
#: Note that copying to the clipboard is a security risk, as all
#: programs, including websites open in your browser can read the
#: contents of the system clipboard.
```

I was not aware of that.
I constantly copy-paste passwords !

Is that really true?
Also for Windows?
I mean, any and all webpages that I have open in a browser?!?
That can be hundreds.

What a strange idea is that?
Why does the OS allow for that?!?

I'm in panic/

**Comment by ket000** [comment]
Note this does not work on rhel 8 or oracle linux 8 due to waylnd issue. Works fine on ubuntu 20.10 focal fossa.

**Comment by JulienPerrin** [comment]
```
mouse_map right press ungrabbed paste_from_clipboard
```
works for me (Wayland Ubuntu 25.10), and I use the middle click to paste from selection without changing the default parameter for it.
